### PR TITLE
module upgrade

### PR DESCRIPTION
**Pull request by nexient-ajoshi**

### CODE REVIEW

**Comment by nexient-ajoshi** [comment]
Terratests need to be fixed. Before making module public, as part of separate PR those will be added.

File: main.tf [hunk 1]
```diff
@@ -10,28 +10,10 @@
...
-  class_env               = var.environment
-  cloud_resource_type     = each.value.name
-  instance_env            = var.environment_number
-  instance_resource       = var.resource_number
-  maximum_length          = each.value.max_length
-  logical_product_family  = var.logical_product_family
-  logical_product_service = var.logical_product_service
-}
-
 module "zone" {
```
Comment (dsahoo-nexient) [inline main.tf:13]: As per our earlier discussions today, this should first be a primitive module and then we can refer from a collection module

**Review by dsahoo-nexient — CHANGES_REQUESTED** [review]

File: main.tf (hunk 1 above)
Comment (nexient-ajoshi) [reply main.tf:13]: I ended up renaming this module. This module is a primitive by itself. It allows to create DNS zones. To create DNS zone records there is separate module.

File: outputs.tf [hunk 2]
```diff
@@ -9,22 +9,27 @@
 // WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 // See the License for the specific language governing permissions and
 // limitations under the License.
-output "zone_id" {
-  value = module.zone.route53_zone_zone_id[var.zone_name]
+output "route53_zone_zone_ids" {
```
Comment (dsahoo-nexient) [inline outputs.tf:12]: Name and the description doesn't match. name is plural and description is singular. Is it going to return multiple IDs or single ID?

File: examples/private_zone/outputs.tf [hunk 3]
```diff
@@ -9,22 +9,27 @@
 // WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 // See the License for the specific language governing permissions and
 // limitations under the License.
-output "zone_id" {
-  value = module.dns_zone.zone_id
+output "route53_zone_zone_ids" {
```
Comment (dsahoo-nexient) [inline examples/private_zone/outputs.tf:12]: same comment as others. description and name doesnt match for all the outputs

File: examples/public_zone/outputs.tf [hunk 4]
```diff
@@ -9,22 +9,27 @@
 // WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 // See the License for the specific language governing permissions and
 // limitations under the License.
-output "zone_id" {
-  value = module.dns_zone.zone_id
+output "route53_zone_zone_ids" {
```
Comment (dsahoo-nexient) [inline examples/public_zone/outputs.tf:12]: name and description doesnt match

**Review by dsahoo-nexient — CHANGES_REQUESTED** [review]

File: examples/private_zone/.terraform.lock.hcl
Comment (chris-taylor-ntt) [inline examples/private_zone/.terraform.lock.hcl:1]: This file is in the .gitignore:

```
# We don't want to commit the test run lock files
.terraform.lock.hcl
```

However, it was probably added to the repo in the same commit as the .gitignore, so it wouldn't have been excluded and now just accumulates changes. Could you remove it from your commit?

File: examples/private_zone/locals.tf [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+    for k, v in var.zones : "dns_zone_${k}" => {
+      name       = "privatezone${index(keys(var.zones), k)}"
+      max_length = 80
+      region     = "us-east-2"
+    }
+  })
+
+  # Merge the tags from the zones variable with the resource name tags. Merge the VPC ID and region with the Zones variable
+  tags = {
+    for idx, v in var.zones : idx => {
```
Comment (chris-taylor-ntt) [inline examples/private_zone/locals.tf:13]: Minor nitpick, but there's a discrepancy between this for loop and the one above it. Above, the map's key is denoted as `k` but here (and below) it's `idx`.

I don't have a strongly-held opinion when it comes to using `idx` or `k` (just a loose preference for `k` given we also have `v`), but I would like to ensure that we're internally consistent both within this file and across the rest of the module. The other example's `locals.tf` has this same issue.

File: variables.tf [hunk 6]
```diff
@@ -10,112 +10,29 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-//variables required by resource names module
-variable "resource_names_map" {
-  description = "A map of key to resource_name that will be used by tf-module-resource_name to generate resource names"
+# Variables required by the dnz zone module
```
Comment (chris-taylor-ntt) [inline variables.tf:13]:
```suggestion
# Variables required by the DNS zone module
```

File: examples/private_zone/locals.tf (hunk 5 above)
Comment (nexient-ajoshi) [reply examples/private_zone/locals.tf]: @chris-taylor-ntt  Thanks Chris!! Fixed it

File: examples/private_zone/.terraform.lock.hcl
Comment (nexient-ajoshi) [reply examples/private_zone/.terraform.lock.hcl:1]: Thanks @chris-taylor-ntt 
Removed it!!

**Review by chris-taylor-ntt — APPROVED** [review]

**Pull request merged**